src/components/sketches/Scene2.ts

```diff
@@ -0,0 +1,148 @@
+import p5 from 'p5';
+import { inEllipse, Point, RainPoint, rand, randInEllipse } from '../../utils';
+import {
+  CLOUD_HEIGHT_MAX,
+  CLOUD_HEIGHT_MIN,
+  CLOUD_OFFSETS,
+  CLOUD_SECTION_OFFSETS,
+  CLOUD_WIDTH_MAX,
+  CLOUD_WIDTH_MIN,
+  NUM_CLOUD_SECTION_PARTICLES,
+  NUM_RAIN,
+  PARTICLE_DIAMETER,
+  RAIN_RADIUS,
+  RAIN_RATE,
+  RAIN_Y_MAX,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+  let clouds: Cloud[];
+  let rain: Rain;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+
+    p.noStroke();
+    p.colorMode(p.HSB, 360);
+    p.frameRate(60);
+
+    rain = new Rain();
+    clouds = CLOUD_OFFSETS.map(({x, y}) =>
+      new Cloud(window.innerWidth / 2 + x, window.innerHeight / 4 + y, rain));
+  };
+
+  p.draw = () => {
+    p.background(20);
+    clouds.forEach((cloud) => {
+      cloud.draw();
+      cloud.checkRain();
+    });
+    rain.draw();
+
+    p.noStroke();
+    p.fill(255);
+    p.circle(p.mouseX, p.mouseY, 25);
+  };
+
+  p.windowResized = () => {
+    p.resizeCanvas(window.innerWidth, window.innerHeight, true);
+  };
+
+  class Cloud {
+    public clouds: _CloudSection[];
+
+    constructor(p_x: number, p_y: number, p_rain: Rain) {
+      this.clouds = CLOUD_SECTION_OFFSETS.map(({x, y}) => {
+        return new _CloudSection(p_x + x, p_y + y, p_rain);
+      });
+    }
+
+    public checkRain() {
+      this.clouds.forEach(c => c.checkRain());
+    }
+
+    public draw() {
+      this.clouds.forEach((c) => {
+        c.draw();
+      });
+    }
+  }
+
+  class Rain {
+    private _rain: RainPoint[];
+
+    constructor() {
+      this._rain = [];
+    }
+
+    addRain(x: number, y: number) {
+      const len = rand(50, 100);
+      const velocity = rand(6, 8);
+      this._rain.push({x, y, len, velocity});
+    }
+
+    draw() {
+      const t_rain: RainPoint[] = [];
```

Comment:
Maybe add a delete rain, once it gets to the bottom of the screen, so that the rainpoint array will not keep increasing as the user keeps hovering.

Reply:
Line 95 does this check!

Reply:
I see I think line 99 with the reassignment was the true check of the data storing the rain being maintained. Actually, I like it cuz it's resetting on every iteration of general p5 draw function. Nice!

Reply:
yup!

Reply:
mmm the reassignment isnt really the check, but it serves the purpose of maintaining the integrity of the rain array without incurring more cost